csrc/id_model/contiguity.cpp

```diff
@@ -123,6 +151,26 @@ void ContigIDGroups::handle(Merge* merge, Direction direction) {
   contig_ids_.emplace(graph_.toGroup(merge->out()));
 }
 
+// Avoid contiguous indexing if going through non-divisible
+// splits. Not all non-divisible splits need specific predicates, so
+// this condition could be relaxed.
+void ContigIDGroups::handle(Split* split, Direction direction) {
+  if (direction == Direction::Forward) {
```

Comment:
Why is this only applied to the forward direction?

Reply:
The problem with split is traversing backward may exceed the extent of the split input. That's why forward splits are checked here (note that this analysis looks at the indexing traversal in a reverse order). When traversing a forward split, there is no possibility of the two output domains to get indices exceeding their respective extents, so that's why only Forward is sufficient here.

Reply:
Makes sense. Thanks for the explanation.